lib/matplotlib/axis.py

```diff
@@ -1725,28 +1725,30 @@ def set_label_position(self, position):
 
     def _update_label_position(self, bboxes, bboxes2):
         """
-        Update the label position based on the sequence of bounding
-        boxes of all the ticklabels
+        Update the label position based on the the bounding box enclosing
+        all the ticklabels and axis spine
         """
         if not self._autolabelpos:
             return
         x, y = self.label.get_position()
         if self.label_position == 'bottom':
-            if not len(bboxes):
-                bottom = self.axes.bbox.ymin
-            else:
-                bbox = mtransforms.Bbox.union(bboxes)
-                bottom = bbox.y0
+            spinebbox = (self.axes.spines['bottom'].get_transform().
```

Comment:
Style alternative: I would factor out `spine = self.axes.spines['bottom']` so as to make the spinebbox expression more compact.